src/index.ts

```diff
@@ -59,74 +181,62 @@ export function wasmBufferDataSlice(buf, offset, length: number) {
 
 // Creates a wasm buffer that can be passed to an svm_codec instance methods from a provided json object
 // Returns the buffer
-export function newWasmBuffer(object: JSON) {
+function newWasmBuffer(object: any) {
     const objectStr = JSON.stringify(object);
     const bytes = new TextEncoder().encode(objectStr);
     const buf = wasmBufferAlloc(bytes.length);
-    assert.strictEqual(bytes.length, wasmBufferLength(buf));
+
+    if (bytes.length !== wasmBufferLength(buf)) {
+        throw new Error("svm codec error: unexpected buf length")
+    }
+
     copyToWasmBufferData(buf, bytes);
     return buf;
 }
 
+
 // Returns a json object of a provided wasm buffer
-export function loadWasmBuffer(buf) : JSON {
+// Todo: write test
+// @ts-ignore
+function loadWasmBuffer(buf) : any {
     let length = wasmBufferLength(buf);
     const slice = wasmBufferDataSlice(buf, 0, length);
     const string = new TextDecoder().decode(slice);
     return JSON.parse(string);
 }
 
+
 // Returns a utf-8 string representation of the data in an svm_codec buffer
 function loadWasmBufferDataAsString(buf) : string {
     let length = wasmBufferLength(buf);
     const slice = wasmBufferDataSlice(buf, 0, length);
-    assert.strictEqual(slice[0], OK_MARKER);
+    if (slice[0] !== OK_MARKER) {
+        throw loadWasmBufferError(buf)
+    }
+
     return new TextDecoder().decode(slice.slice(1));
 }
 
+
 // Returns a json object representation of the data in a svm_codec buffer
 // Throws an exception if buffer has an error with the exception's string representation
-export function loadWasmBufferDataAsJson(buf) : JSON {
+function loadWasmBufferDataAsJson(buf) : JSON {
     let length = wasmBufferLength(buf);
     const slice = wasmBufferDataSlice(buf, 0, length);
 
     if (slice[0] === ERR_MARKER) {
-        const msg = loadWasmBufferError(buf);
-        console.warn(msg);
-        throw msg;
+        throw loadWasmBufferError(buf);
     }
 
-    assert.strictEqual(slice[0], OK_MARKER);
-
     const string = loadWasmBufferDataAsString(buf);
 
     return JSON.parse(string);
 }
 
 // Returns a utf-8 string representation of an error in an svm_codec buffer
-export function loadWasmBufferError(buf) {
+function loadWasmBufferError(buf) {
     let length = wasmBufferLength(buf);
     const slice = wasmBufferDataSlice(buf, 0, length);
     // assert.strictEqual(slice[0], ERR_MARKER);
```

Comment:
I think we want that `assert`

Reply:
@avive

Reply:
Ok but it should return in error string in this case and not throw. Something like this:
if (slice[0] !== ERR_MARKER) {
        return "svm buffer doesn't have an error."
}
return new TextDecoder().decode(slice.slice(1));